Add automated scripts for installfest and update INSTALLFEST.md

scottyscripts opened the pull request:
I automated everywhere that I could and updated the INSTALLFEST.md to reflect changes.

I removed the sections from INSTALLFEST.md for installing rbenv, Ruby, and postgres, but still included the scripts in config/

Fixed Atom autocomplete issue we discussed, will close issue after these changes are merged

scottyscripts committed:
Add automated installfest scripts in config/

scottyscripts committed:
update installfest doc for automated scripts

commands replaced with commands to execute scripts where appropriate
remove Ruby and Postgres sections

scottyscripts committed:
fix syntax errors in installfest doc

gaand commented on config/atom.sh:
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
```
Is there an issue about using cask to install Atom?  Would that make it possible to have a more comprehensive script?  Also, where's a check to see if the commands are available?

jrhorn424 replied:
> Also, where's a check to see if the commands are available?

I agree, this should be added before merge IMO.

scottyscripts replied:
Cask will only work for macOSX. Tested process in Ubuntu machine

gaand commented on config/bash.sh:
```diff
@@ -0,0 +1,27 @@
+    read
+  fi
+
+  echo '/\/usr\/local\/bin/\nd\nwq' | sudo ed /etc/paths
```
There's definitely an issue saying not to do this anymore.

gaand commented on config/git.sh:
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+if which git; then
```
This test is insufficient. Git is installed on macOS by default.

gaand commented on config/npm.sh:
```diff
@@ -0,0 +1,3 @@
+#!/usr/bin/env bash
+npm install --global npm
+npm install --global jshint jsonlint grunt-cli remark-lint jscs bower phantomjs-prebuilt
```
This is not the correct set of global packages.

gaand commented on config/nvm.sh:
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+if uname -s != 'Darwin'; then
+  export NVM_DIR="/home/$(whoami)/.nvm"
```
I believe this is unnecessary if nvm is installed and in the path.

gaand commented on config/nvm.sh:
```diff
@@ -0,0 +1,12 @@
+  export NVM_DIR="/home/$(whoami)/.nvm"
+  [ -s "$NVM_DIR/nvm.sh" ] && . "$NVM_DIR/nvm.sh"
+else
+  export NVM_DIR=~/.nvm
```
As the above

gaand commented on config/postgres.sh:
```diff
@@ -0,0 +1,21 @@
+  sudo -u postgres createdb `whoami`
+else
+  brew install postgres
+  mkdir -p ~/Library/LaunchAgents
```
Is this actually necessary?

gaand commented on config/rbenv.sh:
```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+if which rvm; then
```
Destructive actions should warn the user

scottyscripts replied:
Now prompt user for (y/n). Only remove if yes

gaand commented on config/xcode.sh:
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+if which xcode-select; then
```
I'm not sure why this would work.

gaand reviewed (changes requested):
Why are there separate git scripts?
Are we using linux brew for linux?

Please have a look and let me know if you have questions.

scottyscripts committed:
address issues and update scripts

apm packages updated
git_config.sh now part of git.sh
check if hub is installed with homebrew
npm global packages updated
rbenv.sh now prompts user before removing rvm
install node by version if installing by boron fails
INSTALLFEST.md updated accordingly

scottyscripts commented:
I consolidated git_config.sh with git.sh.

After researching the export NVM_DIR, I continually found issues with nvm command not working unless you do the export

removing rvm now prompts the user for a (y/n)

There are many open issues. I commented and addressed the ones I saw as imperative for Friday, but will be clearing the rest.

scottyscripts committed:
update if statements checking OS

scottyscripts committed:
combine git configuration scripts

automate ssh-keygen and linking/testing with github and combine scripts
update INSTALLFEST.md to reflect changes

scottyscripts committed:
fix syntax errors in scripts
all scripts working
remove xcode script
Update INSTALLFEST.md with new commands

scottyscripts commented:
Tested all scripts, fixing any errors encountered. Consolidated two of the git scripts. Looked more into installing atom with cask, and I'm not sure that it would speed up the process. Also edited the INSTALLFEST.md doc to reflect changes.

scottyscripts committed:
INSTALLFEST doc order swapped around

move git section to beginning so devs can clone repo with scripts
git_ignore.sh config combined with git.sh

scottyscripts committed:
fix typos in INSTALLFEST.md doc

jrhorn424 commented on INSTALLFEST.md:
```diff
@@ -266,411 +164,80 @@ brew install nvm
-> features and commands that make working with GitHub easier. -- [`hub`
-> README](https://github.com/github/hub).
+1.  Download Atom from [Atom.io](https://atom.io/).
+1.  Move Atom from downloads to the Application directory.
```
This step is very important. Without it, the command line tool installation will claim to work when it does not actually work. This is due to sandboxing on macOS.

scottyscripts replied:
I moved these steps and the atom part to end of installfest. cask will work to install on mac, but not linux

jrhorn424 commented on config/nvm.sh:
```diff
@@ -0,0 +1,12 @@
+  source $(brew --prefix nvm)/nvm.sh
+fi
+
+nvm install --lts=boron || nvm install 6.9.4
```
Boron is now at 6.10.2. I'm not sure what we want to do about that @gaand.

scottyscripts committed:
confirm apm is available before atom script runs

gaand commented on config/rbenv.sh:
```diff
@@ -0,0 +1,47 @@
+# if you do not get the message 'rbenv 1.1.0' refer to an instructor \n
+Press [ENTER] to continue"
+
+echo bundler > ~/.rbenv/default-gems
```
Destructive.

gaand commented on config/rbenv.sh:
```diff
@@ -0,0 +1,47 @@
+fi
+
+rbenv --version
+read -p "# expecting message 'rbenv 1.1.0'
```
Check this directly, don't prompt

scottyscripts replied:
This is now automated

gaand reviewed (changes requested):
Linux brew is important.

scottyscripts committed:
use linux brew if not on linux

NOTE: we have not been doing this on installfest prior

scottyscripts committed:
automate checking rbenv version in rbenv.sh

scottyscripts committed:
move linux brew installation to homebrew.sh

for some reason linux brew was being installed at a different time than
mac brew

scottyscripts committed:
script execution order moved

hub.sh requires brew to run

scottyscripts committed:
now install atom with cask on OSX

if apm command not found, terminal prompts to install via Atom GUI

scottyscripts commented:
Looked into cask for installing atom, but it only works on mac. I updated INSTALLFEST.md to tell them to run `brew cask install atom` for mac (kept old command for linux). I will test scripts again tonight on linux and mac just to make sure we're all good for tomorrow. 

Let me know if anything is unclear.
